src/sage/rings/number_field/order.py

```diff
@@ -2082,7 +2082,16 @@ def _repr_(self):
                     s = "Gaussian Integers"
         else:
             s = "Order"
-        return s + " in " + repr(self._K)
+            if self.number_field().absolute_degree() == 2:
+                f = (self.discriminant() / self.number_field().discriminant()).sqrt()
+                s += f' of conductor {f}'
+        try:
```

Comment:
For separation of concerns, you may put the conductor formula into the method `self.conductor()` and just use the method here?